warplib/memory.cairo

```diff
@@ -35,367 +21,346 @@ from warplib.maths.utils import felt_to_uint256, narrow_safe
 
 // -----------------Scalars-----------------
 
-func wm_read_felt{warp_memory: DictAccess*}(loc: felt) -> (val: felt) {
-    let (res) = dict_read{dict_ptr=warp_memory}(loc);
-    return (res,);
+use core::dict::DictFeltToTrait;
+use core::integer;
+use core::integer::u128_from_felt;
+use core::integer::u128_to_felt;
+use core::integer::u256_from_felt;
+use core::array::ArrayTrait;
+
+extern fn narrow_safe(x: u256) -> felt nopanic; // can be replaced with import once available
+
+
+fn wm_read_felt(ref warp_memory: DictFeltTo::<felt> ,  loc: felt) -> felt {
+    warp_memory.get(loc)
 }
 
-func wm_read_256{warp_memory: DictAccess*}(loc: felt) -> (val: Uint256) {
-    let (low) = dict_read{dict_ptr=warp_memory}(loc);
-    let (high) = dict_read{dict_ptr=warp_memory}(loc + 1);
-    return (Uint256(low, high),);
+fn wm_read_256(ref warp_memory: DictFeltTo::<felt> ,  loc: felt) -> u256 {
```

Comment:
I think `warp_memory` should remain an implicit, if we are doing a direct translation, otherwise it would require extensive changes in multiple places in the code